examples/qml_features/rust/src/nested_qobjects.rs

```diff
@@ -40,6 +45,25 @@ mod ffi {
     }
 
     impl qobject::OuterObject {
+        #[qinvokable]
+        pub fn initialise(self: Pin<&mut Self>) {
+            // Example of connecting a signal from one QObject to another QObject
+            // this causes OuterObject::Called to trigger InnerObject::Called
+            self.on_called(
+                |qobject, obj| {
+                    // We need to convert the *mut T to a Pin<&mut T> so that we can reach the methods
+                    if let Some(inner) = unsafe { qobject.inner().as_mut() } {
+                        // TODO: Use `pin!` one it's stable so that this unsafe block can be removed
+                        // https://doc.rust-lang.org/std/pin/macro.pin.html
```

Comment:
This is stable now.

Reply:
Yup but unfortunately doesn't quite help us enough, it can go from T -> Pin<&mut T>, but we need to get from *mut T to T somehow. This is just copied from the other code below and should be solved in another pull request (i have another branch locally with experiments).

Reply:
(As ultimately we need to go from *mut T -> Pin<&mut T>

Reply:
Removed this comment for now as it doesn't help us.